File: ProjectJava/src/main/java/com/example/projectjava/InterfaceRepositories/CharityCaseRepository.java
package com.example.projectjava.InterfaceRepositories;


import com.example.projectjava.Domain.CharityCase;

public interface CharityCaseRepository extends Repository<Integer, CharityCase> {

    void updateSumOfTotalDonations(int idCharityCase,int donation) throws Exception;

    void removeDonation(int id,int donation) throws Exception;
}

File: ProjectC#/WindowsFormsApp/WindowsFormsApp/InterfaceRepository/CharityCaseRepository.cs
using ConsoleApp.Domain;
using ConsoleApp.Repository;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp.InterfaceRepository
{
    public interface CharityCaseRepository: Repository<int,CharityCase>
    {
        void addDonation(int idCharityCase, int donation);
        void removeDonation(int idCharityCase, int donation);
    }
}
